tests/test_media_urlencoded.py

```diff
@@ -0,0 +1,71 @@
+import io
+
+import pytest
+
+import falcon
+from falcon import media
+from falcon import testing
+
+
+def test_deserialize_empty_form():
+    handler = media.URLEncodedFormHandler()
+    stream = io.BytesIO(b'')
+    assert handler.deserialize(stream, falcon.MEDIA_URLENCODED, 0) is None
+
+
+def test_deserialize_invalid_unicode():
+    handler = media.URLEncodedFormHandler()
+    stream = io.BytesIO('spade=♠'.encode())
+    with pytest.raises(UnicodeDecodeError):
+        print(handler.deserialize(stream, falcon.MEDIA_URLENCODED, 9))
+
+
+@pytest.mark.parametrize('data,expected', [
+    ({'hello': 'world'}, 'hello=world'),
+    ({'number': [1, 2]}, 'number=1&number=2'),
+])
+def test_urlencoded_form_handler_serialize(data, expected):
+    handler = media.URLEncodedFormHandler()
+    assert handler.serialize(data, falcon.MEDIA_URLENCODED) == expected
+
+
+class MediaMirror:
+
+    @classmethod
```

Comment:
Perhaps I'm overlooking something, but why do we need the `classmethod`?

Reply:
No specific reason tbh, it could well be a `staticmethod` or a normal class method with `self`. I think I was planning to subclass it, but that part never got finished.

I can change to a normal method.

Reply:
@vytas7 I'd prefer to use `staticmethod` decorator here, it recommended to use for functions when `self` haven't used explicitly.

IMHO.

Reply:
@vytas7 I'd prefer to use `staticmethod` decorator here, it recommended to use for functions when `self` haven't used explicitly.

IMHO.

Reply:
@mikeylight it's a matter of taste, if you use class variables (which we don't do here), I prefer to define a `@classmethod`. I'll change this one to a normal method with `self` though not to complicate the matters too much.